view/infinite.go

```diff
@@ -0,0 +1,158 @@
+package view
+
+import (
+	"errors"
+	"fmt"
+	"math"
+	"time"
+
+	"github.com/jsdelivr/globalping-cli/client"
+	"github.com/jsdelivr/globalping-cli/model"
+	"github.com/pterm/pterm"
+)
+
+func OutputInfinite(id string, ctx *model.Context) error {
+	fetcher := client.NewMeasurementsFetcher(client.ApiUrl)
+	res, err := fetcher.GetMeasurement(id)
+	if err != nil {
+		return err
+	}
+	// Probe may not have started yet
+	for len(res.Results) == 0 {
+		time.Sleep(apiPollInterval)
+		res, err = fetcher.GetMeasurement(id)
+		if err != nil {
+			return err
+		}
+	}
+	// Wait for results to be complete
+	for res.Status == "in-progress" {
+		time.Sleep(apiPollInterval)
+		res, err = fetcher.GetMeasurement(res.ID)
+		if err != nil {
+			return err
+		}
+	}
+	if ctx.Latency {
+		return OutputLatency(id, res, *ctx)
+	}
+
+	if ctx.JsonOutput {
+		return OutputJson(id, fetcher, *ctx)
+	}
+
+	if len(res.Results) == 1 {
+		return outputSingleLocation(res, ctx)
+	}
+	return outputMultipleLocations(res, ctx)
+}
+
+func outputSingleLocation(res *model.GetMeasurement, ctx *model.Context) error {
+	measurement := &res.Results[0]
+	if len(ctx.Stats) == 0 {
+		// Initialize state
+		ctx.Stats = make([]model.MeasurementStats, 1)
+		// Print header
+		fmt.Println(generateHeader(measurement, !ctx.CI))
+		fmt.Printf("PING %s (%s) 56(84) bytes of data.\n", res.Target, measurement.Result.ResolvedAddress)
+	}
+	timings, err := client.DecodePingTimings(measurement.Result.TimingsRaw)
+	if err != nil {
+		return err
+	}
+	for i := range timings {
+		ctx.Stats[0].Sent++
+		t := timings[i]
+		fmt.Printf("64 bytes from %s (%s): icmp_seq=%d ttl=%d time=%.2f ms\n",
+			measurement.Result.ResolvedHostname,
+			measurement.Result.ResolvedAddress,
+			ctx.Stats[0].Sent,
+			t.TTL,
+			t.RTT)
+	}
+	return nil
+}
+
+func outputMultipleLocations(res *model.GetMeasurement, ctx *model.Context) error {
+	var err error
+	if len(ctx.Stats) == 0 {
+		// Initialize state
+		ctx.Stats = make([]model.MeasurementStats, len(res.Results))
+		for i := range ctx.Stats {
+			ctx.Stats[i].Min = math.MaxFloat64
+		}
+		// Create new writer
+		ctx.Area, err = pterm.DefaultArea.Start()
+		if err != nil {
+			return errors.New("failed to start writer: " + err.Error())
+		}
+	}
+	tableData := pterm.TableData{
+		{"Location", "Loss", "Sent", "Last", "Avg", "Min", "Max"},
+	}
+	for i := range res.Results {
+		result := &res.Results[i]
+		localStats := &ctx.Stats[i]
+		updateMeasurementStats(localStats, result)
+		tableData = append(tableData, []string{
+			getLocationText(result),
+			formatValue(fmt.Sprintf("%.2f", localStats.Loss)+"%", 6),
+			formatValue(fmt.Sprintf("%d", localStats.Sent), 3),
```

Comment:
This seems to have two whitespace characters on the right instead of one for some reason.